src/Microsoft.ML.Data/DataView/BatchDataViewMapperBase.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Data.DataView
+{
+    internal abstract class BatchDataViewMapperBase<TInput, TBatch> : IDataView
+    {
+        public bool CanShuffle => false;
+
+        public DataViewSchema Schema => SchemaBindings.AsSchema;
+
+        private readonly IDataView _source;
+        private readonly IHost _host;
```

Comment:
>_host [](start = 31, length = 5)

Make this protected, and use in the deriving class.

Reply:
fixed.

---
In reply to: [426397354](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426397354) [](ancestors = 426397354)